atlasdb-init-api/src/main/java/com.palantir.atlasdb.init/InitializationCheckerProxy.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.init;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javassist.util.proxy.MethodHandler;
+import javassist.util.proxy.Proxy;
+import javassist.util.proxy.ProxyFactory;
+
+public final class InitializationCheckerProxy {
+    private static Logger log = LoggerFactory.getLogger(InitializationCheckerProxy.class);
+
+    private InitializationCheckerProxy() {
+        //factory
+    }
+
+    public static  <T> T newProxyInstance(Class<?>[] classes,
+            Object[] objects,
+            Class<T> clazz)
+            throws InvocationTargetException, NoSuchMethodException, InstantiationException, IllegalAccessException {
```

Comment:
instead of using javassist, we should probably just extract an interface from CassandraClientPool and use regular Java proxy to avoid another 3rd party library dependency, and allow for better testing/mocking of the pool